### PR TITLE
[Issue 64] Do not use internal SparkSQL type conversion

### PR DESCRIPTION
**Pull request by falaki**
We were relying on utility classes under ```org.apache.spark.sql.catalyst.expressions._```. This patch removes all those dependancies.

cc @pwendell and @rxin 

closes issue #64

### CODE REVIEW

**Comment by yhuai** [comment]
I guess creating those converters as functions first and then apply them to the RDD of rows will give us better performance (like what do we at [here] (https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala#L486-L488). Basically, instead of using `CatalystTypeConverters.createToCatalystConverter`, we create a string to scala converters). But, probably it should be in another pr.

File: src/main/scala/com/databricks/spark/csv/util/TypeCast.scala [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      case IntegerType => datum.toInt
+      case LongType => datum.toLong
+      case FloatType => datum.toFloat
+      case DoubleType => datum.toDouble
+      case BooleanType => datum.toBoolean
+      // TODO(hossein): would be good to support other common timestamp formats
+      case TimestampType => Timestamp.valueOf(datum)
+      // TODO(hossein): would be good to support other common date formats
+      case DateType => Date.valueOf(datum)
+      case StringType => datum
```
Comment (yhuai) [inline src/main/scala/com/databricks/spark/csv/util/TypeCast.scala]: Seems we may break existing use cases if we do not support `DecimalType`? I think we can create a `java.math.BigDecimal` directly from the string.

**Comment by rxin** [comment]
@yhuai I wouldn't worry too much about performance here. CSV parsing is super slow anyway :)

**Comment by shivaram** [comment]
FWIW I just ran into this bug while trying to apply a schema from SparkR.

**Comment by falaki** [comment]
@yhuai Addressed comments. ptal.

File: src/test/scala/com/databricks/spark/csv/util/TypeCastSuite.scala [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  test("Can parse decimal type values") {
+    val stringValues = Seq("10.05", "1,000.01", "158,058,049.001")
+    val decimalValues = Seq(10.05, 1000.01, 158058049.001)
+    val decimalType = new DecimalType(None)
+
+    stringValues.zip(decimalValues).foreach { case (strVal, decimalVal) =>
+      assert(TypeCast.castTo(strVal, decimalType) === new BigDecimal(decimalVal.toString))
+    }
+  }
+}
```
Comment (yhuai) [inline src/test/scala/com/databricks/spark/csv/util/TypeCastSuite.scala]: new line at here?

**Comment by yhuai** [comment]
just a tiny format thing. LGTM

**Pull request merged**